src/ucp/core/ucp_mm.c

```diff
@@ -192,6 +192,12 @@ static ucs_status_t ucp_mem_alloc(ucp_context_h context, size_t length,
     {
         method = context->config.alloc_methods[method_index].method;
 
+        /* Non-host memory allocation supported on METHOD_MD alone */
+        if ((memh->mem_type != UCS_MEMORY_TYPE_HOST)
+            && (method != UCT_ALLOC_METHOD_MD)) {
+            continue;
+        }
```

Comment:
what happens if we remove? it should be done already in UCT layer

Reply:
removed